gobblin-config-management/src/main/java/gobblin/config/configstore/impl/SimpleConfigStoreResolver.java

```diff
@@ -0,0 +1,131 @@
+package gobblin.config.configstore.impl;
+
+import gobblin.config.configstore.ConfigStore;
+import gobblin.config.configstore.ConfigStoreWithResolution;
+import gobblin.config.utils.CircularDependencyChecker;
+import gobblin.config.utils.PathUtils;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValue;
+
+
+public class SimpleConfigStoreResolver implements ConfigStoreWithResolution {
+
+  private static final Logger LOG = Logger.getLogger(SimpleConfigStoreResolver.class);
+  private final ConfigStore store;
+
+  public SimpleConfigStoreResolver(ConfigStore base) {
+    this.store = base;
+  }
+
+  @Override
+  public String getCurrentVersion() {
+    return store.getCurrentVersion();
+  }
+
+  @Override
+  public URI getStoreURI() {
+    return store.getStoreURI();
+  }
+
+  @Override
+  public Collection<URI> getChildren(URI uri, String version) {
+    return store.getChildren(uri, version);
+  }
+
+  @Override
+  public Collection<URI> getOwnImports(URI uri, String version) {
+    return store.getOwnImports(uri, version);
+  }
+
+  @Override
+  public Config getOwnConfig(URI uri, String version) {
+    return store.getOwnConfig(uri, version);
+  }
+
+  @Override
+  public Config getResolvedConfig(URI uri, String version) {
+    UUID traceID = UUID.randomUUID();
+    LOG.info(String.format("Getting resolved config for uri: %s for version %s, with traceID: %s", 
+        uri, version, traceID));
+    return getResolvedConfigWithTrace(uri, version, traceID.toString());
+  }
+
+  protected Config getResolvedConfigWithTrace(URI uri, String version, String traceID) {
```

Comment:
The client may query multiple URIs and even multiple times for the same URI. traceID is used to annotate the single session of the query. Easy to find the whole query in Log